cirq-core/cirq/ops/classically_controlled_operation_test.py

```diff
@@ -259,6 +259,43 @@ def test_key_set(sim):
     assert result.measurements['b'] == 1
 
 
+@pytest.mark.parametrize('sim', ALL_SIMULATORS)
+def test_repeated_measurement_unset(sim):
+    q0, q1 = cirq.LineQubit.range(2)
+    circuit = cirq.Circuit(
+        cirq.measure(q0, key='a'),
+        cirq.X(q0),
+        cirq.measure(q0, key='a'),
+        cirq.X(q1).with_classical_controls(cirq.KeyCondition(cirq.MeasurementKey('a'), index=-2)),
+        cirq.measure(q1, key='b'),
+        cirq.X(q1).with_classical_controls(cirq.KeyCondition(cirq.MeasurementKey('a'), index=-1)),
+        cirq.measure(q1, key='c'),
+    )
+    result = sim.run(circuit)
+    assert result.measurements['a'] == 1
```

Comment:
Okay, yeah I don't think there's anything I can do on my own. Have to wait for that PR to go in, then fix up `simulator.run` to initialize `Result` with the full histories as part of this PR.

Reply:
Or we could do it in the opposite order too. Just, whichever order we do it, the second one should fix up `simulator.run` accordingly.

Reply:
I favor merging #4555 first, if only to avoid a weird intermediate state with this one. In any case, both PRs will need to go in before the v0.14 release.